src/nvim/eval.c

```diff
@@ -5421,6 +5421,9 @@ static int list_join(garray_T *gap, list_T *l, char_u *sep, int echo_style, int
   garray_T join_ga;
   int retval;
 
+  if (l->lv_len < 1)
+    return OK;
```

Comment:
> but it's a pretty widely used convention and the documentation can be changed to make it explicit.

Yes, but I don't think we should be consistent for the sake of consistency, especially since `true/false` loses information.